plugins/pay.c

```diff
@@ -1979,6 +1990,30 @@ payment_listsendpays_previous(struct command *cmd, const char *buf,
 			completed = false;
 			pending = false;
 			last_group = groupid;
+
+			parts = 1;
+			json_scan(tmpctx, buf, t,
```

Comment:
Hmm, is there no group_id 0?  Since you initialize last_group to 0, it seems you won't run this for the first group.  Since msat is not initialized elsewhere, this means you will use it uninitialized below?

Reply:
Could either make it a pointer to make that explicit, but the msats field is only used if we detect that there was a prior success or failure and replicate the result for idempotency, so not having a prior attempt means that we'll not use msats at al